cosmo_tester/test_suites/bootstrap_based_tests/cli_test.py

```diff
@@ -38,112 +38,101 @@
 
 
 @pytest.fixture(scope='function')
-def cli_package_tester(ssh_key, attributes, tmpdir, logger):
-    logger.info('Using temp dir: %s', tmpdir)
-    tmpdir = Path(tmpdir)
-
-    tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _CliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
-
-    yield tester
-
-    tester.perform_cleanup()
-
-
-@pytest.fixture(scope='function')
-def windows_cli_package_tester(ssh_key, attributes, tmpdir, logger):
-    logger.info('Using temp dir: %s', tmpdir)
-    tmpdir = Path(tmpdir)
-
-    tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _WindowsCliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
-
-    yield tester
-
-    tester.perform_cleanup()
-
+def package_tester(request, ssh_key, attributes, tmpdir, logger):
+    _package_tester_mapping = {
+        'linux': _CliPackageTester,
+        'osx': _OSXCliPackageTester,
+        'windows': _WindowsCliPackageTester
+    }
+    platform = request.param
+    tester_class = _package_tester_mapping[platform]
 
-@pytest.fixture(scope='function')
-def osx_cli_package_tester(ssh_key, attributes, tmpdir, logger):
     logger.info('Using temp dir: %s', tmpdir)
     tmpdir = Path(tmpdir)
 
     tf_inputs = get_terraform_inputs(attributes, ssh_key)
-    tester = _OSXCliPackageTester(tmpdir, tf_inputs, ssh_key, logger)
+    tester = tester_class(tmpdir, tf_inputs, ssh_key, logger)
 
     yield tester
 
     tester.perform_cleanup()
 
 
-def test_cli_on_centos_7(cli_package_tester, attributes):
-    cli_package_tester.inputs.update({
+@pytest.mark.parametrize('package_tester', ['linux'], indirect=True)
+def test_cli_on_centos_7(package_tester, attributes):
+    package_tester.inputs.update({
         'cli_image': attributes.centos_7_image_name,
         'cli_user': attributes.centos_7_username,
         'manager_image': attributes.centos_7_image_name,
         'manager_user': attributes.centos_7_username,
         'cli_package_url': get_cli_package_url('rhel_centos_cli_package_url')
     })
-    cli_package_tester.run_test()
+    package_tester.run_test()
```

Comment:
kinda sucks that we have the 2-step "update inputs + `.run_test()`". Looks like we should be able to make it `.run_test(inputs)`